src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -68,107 +68,199 @@ def validate_location(namespace):
+            error = '{}\n\tnot applicable: {}'.format(error, forbidden_string)
+        raise CLIError(error)
+
+
+def _validate_managed_disk_sku(sku):
+
+    allowed_skus = ['Premium_LRS', 'Standard_LRS']
+    if sku.lower() not in [x.lower() for x in allowed_skus]:
+        raise CLIError("invalid storage SKU '{}': allowed values: '{}'".format(sku, allowed_skus))
+
+
+# pylint: disable=too-many-branches, too-many-statements, redefined-variable-type
+def _validate_vm_create_storage_profile(namespace, for_scale_set=False):
+
+    # use minimal parameters to resolve the expected storage profile
+    if getattr(namespace, 'attach_os_disk', None) and not namespace.image:
+        if namespace.use_unmanaged_disk:
+            # STORAGE PROFILE #3
+            namespace.storage_profile = StorageProfile.SASpecializedOSDisk
         else:
-            # last try: is it a custom image name?
-            compute_client = _compute_client_factory()
-            try:
-                compute_client.images.get(namespace.resource_group_name, image)
-                image = namespace.image = _get_resource_id(image, namespace.resource_group_name,
-                                                           'images', 'Microsoft.Compute')
-                namespace.storage_profile = StorageProfile.ManagedCustomImage
-            except CloudError:
-                err = 'Invalid image "{}". Use a custom image name, id, or pick one from {}'
-                raise CLIError(err.format(image, [x['urnAlias'] for x in images]))
+            # STORAGE PROFILE #6
+            namespace.storage_profile = StorageProfile.ManagedSpecializedOSDisk
+    elif namespace.image and not getattr(namespace, 'attach_os_disk', None):
+        image_type = _parse_image_argument(namespace)
+        if image_type == 'uri':
+            # STORAGE PROFILE #2
+            namespace.storage_profile = StorageProfile.SACustomImage
+        elif image_type == 'image_id':
+            # STORAGE PROFILE #5
+            namespace.storage_profile = StorageProfile.ManagedCustomImage
+        elif image_type == 'urn':
+            if namespace.use_unmanaged_disk:
+                # STORAGE PROFILE #1
+                namespace.storage_profile = StorageProfile.SAPirImage
+            else:
+                # STORAGE PROFILE #4
+                namespace.storage_profile = StorageProfile.ManagedPirImage
+        else:
+            raise CLIError('Unrecognized image type: {}'.format(image_type))
+    else:
+        # did not specify image XOR attach-os-disk
+        raise CLIError('incorrect usage: --image IMAGE | --attach-os-disk DISK')
+
+    # perform parameter validation for the specific storage profile
+    # start with the required/forbidden parameters for VM
+    if namespace.storage_profile == StorageProfile.ManagedPirImage:
+        required = ['image']
+        forbidden = ['os_type', 'attach_os_disk', 'storage_account',
+                     'storage_container_name', 'use_unmanaged_disk']
```

Comment:
Right. It's here more for documentation purposes. If in the future the logic to quickly guess the profile is removed, this would still remain accurate.